### DISCUSSION TITLE
qdrant_client.get_fastembed_vector_params() with upload_collection

### DISCUSSION THREAD

**Discussion by uguraydrd** [qdrant/qdrant-client#500] [Q&A]
Hello folks!

Firstly, I would like to run fastembed integration. I suppose there is a bug or problem or I am wrong. I am going to explain this problem.

Normally, you know we follow these steps for example if we want to give the vector size:

```
qdrant_client = QdrantClient("http://localhost:6333")
qdrant_client.set_model("BAAI/bge-small-en-v1.5")
qdrant_client.recreate_collection(
    collection_name="test",
    vectors_config=VectorParams(size=384, distance=Distance.COSINE),
)
```
We know there is no problem here. 

But if we want the vector size to be retrieved dynamically based on the model we follow these steps as in documentation like:

```
qdrant_client = QdrantClient("http://localhost:6333")
qdrant_client.set_model("BAAI/bge-small-en-v1.5")
qdrant_client.recreate_collection(
    collection_name="test",
    vectors_config=qdrant_client.get_fastembed_vector_params()
)
```
unfortunately my vectors are not loaded into the collection. I do not get any error but my vectors are not loaded either.

We know `vectors_config` has two typehint like `Union[types.VectorParams, Mapping[str, types.VectorParams]]`. We also have a mapping. After calling `qdrant_client.get_fastembed_vector_params()` , it returns a dictionary for us to feed to `vectors_config` like this:

```
{
    "fast-bge-small-en-v1.5": models.VectorParams(
        size = embeddings_size,
        distance = distance,
        ...
    )
}
```

Finally,

I check the your dev codes. Based on my many tests and observations, I noticed if I do this there is no problem but this is not a solution, just an indication that it works if I do this:

```
qdrant_client = QdrantClient("http://localhost:6333")
qdrant_client.set_model("BAAI/bge-small-en-v1.5")
qdrant_client.recreate_collection(
    collection_name="test",
    vectors_config=qdrant_client.get_fastembed_vector_params()[self.client.get_vector_field_name()]
)
```
because if I do this 'vectors_config' is no longer a dictionary but a pydantic class and equals a value like below. It can then be verified directly by `models.VectorParams`.

`VectorParams(size=384, distance=<Distance.COSINE: 'Cosine'>, ...)`

What is your comment on this topic? Please let me know what is the right approach or solution?

Cheers!

--
Uguray

**Comment by joein** [comment]
Hi @uguraydrd 

Could you show please how you upload points to your collection?

**Comment by uguraydrd** [comment]
Hi @joein,

Of course. I do this:

```
metadata = data["col"].to_dict(orient="records")
model = qdrant_client.embedding_models[qdrant_client.embedding_model_name]

qdrant_client.upload_collection(
    collection_name="test",
    vectors=[
        arr.tolist()
        for arr in list(
            model.passage_embed(
                texts=data,
                batch_size=128,
            )
        )
    ],
    ids=range(len(data)),
    payload=metadata,
    batch_size=128,
)
```

**Comment by joein** [comment]
Fastembed creates a named vector with this name `fast-bge-small-en-v1.5` in your case, however you are uploading unnamed vectors, that's why you they are not added to your collection.

You don't see any error on python client side, because upload_collection has param `wait` set to `False` by default, which means that client does not wait for the response from the server.

Here is a [link](https://qdrant.tech/documentation/concepts/points/?selector=aHRtbCA%2BIGJvZHkgPiBkaXY6bnRoLW9mLXR5cGUoMSkgPiBzZWN0aW9uID4gZGl2ID4gZGl2ID4gZGl2Om50aC1vZi10eXBlKDIpID4gYXJ0aWNsZSA%2BIHA6bnRoLW9mLXR5cGUoMjgp#:~:text=If%20the%20collection%20was%20created%20with%20multiple%20vectors%2C%20each%20vector%20data%20can%20be%20provided%20using%20the%20vector%E2%80%99s%20name%3A) to an example with `upsert`

You can check accepted param types of vectors in `upload_collection` signature.
You can also find examples of usage it with various input types in [tests](https://github.com/qdrant/qdrant-client/blob/master/tests/congruence_tests/test_updates.py)

**Comment by uguraydrd** [comment]
Thank you very much for your information. Finally, It works for me, but normally, I could not find or see this anything on your tutorials unfortunately . When I analyzed `qdrant_client.add()` deeply, I noticed the situation you mentioned.

I want to put the working code here, maybe it will be useful to other people.

```
metadata = data["col"].to_dict(orient="records")
model = qdrant_client.embedding_models[qdrant_client.embedding_model_name]
vector_field_name = qdrant_client.get_vector_field_name()

qdrant_client.upload_collection(
    collection_name="test",
    vectors=[
        {self.vector_field_name: arr.tolist()}
        for arr in self.model.passage_embed(
            texts=data,
            batch_size=128,
        )
    ],
    ids=range(len(data)),
    payload=metadata,
    batch_size=128,
)
```
